db.go

```diff
@@ -332,7 +332,12 @@ func (db *DB) mmap(minsz int) error {
 
 	// Memory-map the data file as a byte slice.
 	if err := mmap(db, size); err != nil {
-		return err
```

Comment:
the previous returned error will cause the db library to panic? if that is the case, shouldnt we fix that behavior instead?

Reply:
like if db fails on mmap, then all following calls into it should return mmap failure error instead of causing pancis?

Reply:
As far as I can tell, the reason is panics after returning the error is that calling `unmap` invalidates `db.meta0` and/or `db.meta1`. So we could also fix this by modifying the logic to never access those variables if `mmap` fails, and put the db in an "error" state, as you suggested.

That's probably a better solution than attempting to remap the old db. Just recently [we received a bug report](https://github.com/NebulousLabs/Sia/issues/2376) where it seems the remap failed as well, causing a panic. The one advantage of remapping is that you can still read from the db, you just can't write any more. But seeing as it's a failure state anyway, I don't think being able to read the db adds a ton of value.

Reply:
@lukechampine 

You are right. I am fine with your current solution as I mentioned. But, I want to make sure this does fix the original issue. Also we need to have sort of a test for this ideally.

thank you!